components/amount-buttons.jsx

```diff
@@ -110,9 +139,9 @@ var AmountButtons = React.createClass({
           </div>
           <div className="two-third">
             <div id="amount-other-container">
-              <input type="radio" name="donation_amount" value="other" id="amount-other" data-parsley-multiple="donation_amount" data-parsley-group="page-1"/>
+              <input type="radio" name="donation_amount" value={this.state.otherValue} id="amount-other" data-parsley-multiple="donation_amount" data-parsley-group="page-1"/>
```

Comment:
I start to wonder if you should change this to `linkState` instead and whenever or wherever you're trying to use it you can always pull it from the `state` object instead.